File: Examples/Ruby/Email/add_new_email.rb
#ExStart:
require 'aspose_email_cloud'

class Email

  include AsposeEmailCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("", "")
    @email_api = EmailApi.new
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../../../data/" << file_name,"r") { |io| io.read } )
  end

  # Add new email.
  def add_new_email
    file_name = "email_test2.eml"
    upload_file(file_name)

    email_document = EmailDocument.new

    email_body = EmailProperty.new
    email_to = EmailProperty.new
    email_from = EmailProperty.new

    email_body.name = "Body"
    email_body.value = "This is body"

    email_to.name = "To"
    email_to.value = "developer@aspose.com"

    email_from.name = "From"
    email_from.value = "sales@aspose.com"

    email_properties = EmailProperties.new
    email_properties.list = [email_body, email_to, email_from]

    email_document.document_properties = email_properties
    email_document.format = "eml"

    response = @email_api.put_create_new_email(file_name, email_document)
  end

end

email = Email.new()
puts email.add_new_email
#ExEnd:

File: Examples/Perl/Email/AddNewEmail.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;

use AsposeEmailCloud::EmailApi;
use AsposeEmailCloud::ApiClient;
use AsposeEmailCloud::Configuration;

use AsposeEmailCloud::Object::EmailProperty;
use AsposeEmailCloud::Object::EmailProperties;
use AsposeEmailCloud::Object::EmailDocument;

my $configFile = '../Config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposeEmailCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeEmailCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeEmailCloud::Configuration::debug = 1;

$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Email API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $emailApi = AsposeEmailCloud::EmailApi->new();

# Set input file name
my $name = 'email_test2.eml';
	
my @emailProperty1 = AsposeEmailCloud::Object::EmailProperty->new('Name' => 'Body', 'Value' => 'This is a body');
my @emailProperty2 = AsposeEmailCloud::Object::EmailProperty->new('Name' => 'To', 'Value' => 'developer@aspose.com');
my @emailProperty3 = AsposeEmailCloud::Object::EmailProperty->new('Name' => 'From', 'Value' => 'sales@aspose.com');
my $emailProperties = AsposeEmailCloud::Object::EmailProperties->new('List' => [@emailProperty1, @emailProperty2, @emailProperty3]);
my $emailDocument = AsposeEmailCloud::Object::EmailDocument->new('DocumentProperties' => $emailProperties, 'Format' => 'eml');

# Upload file to aspose cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name);

# Invoke Aspose.Email Cloud SDK API to add new email                         
$response = $emailApi->PutCreateNewEmail(name=> $name, body=>$emailDocument);

if($response->{'Status'} eq 'OK'){
	# Download email from cloud storage 
	my $output_file = $out_path. $name;
	$response = $storageApi->GetDownload(Path => $name);;
	write_file($output_file, { binmode => ":raw" }, $response->{'Content'});
}
#ExEnd:1